src/main/java/org/jabref/model/database/shared/DBMSType.java

```diff
@@ -51,11 +51,15 @@ public int getDefaultPort() {
     }
 
     public static Optional<DBMSType> fromString(String typeName) {
-        try {
-            return Optional.of(Enum.valueOf(DBMSType.class, typeName.toUpperCase(Locale.ENGLISH)));
-        } catch (IllegalArgumentException exception) {
+        if (typeName == null) {
             return Optional.empty();
+        } else {
+            try {
```

Comment:
Using `DBMSType.values().stream().filter(...equalsIgnoreCase...).findFirst()` is more readable in my opinion.